app/models/user.rb

```diff
@@ -1,18 +1,31 @@
 # frozen_string_literal: true
 
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable, :registerable, and :omniauthable
+  GITHUB_ORGS_QUERY_URL = URI.parse('https://api.github.com/user/orgs')
+
   devise :rememberable, :omniauthable, omniauth_providers: [:github]
 
   validates :login, :uid, presence: true, uniqueness: true
   validates :email, :uid, presence: true, uniqueness: true
 
   def self.from_omniauth(auth)
-    where(provider: auth.provider, uid: auth.uid).first_or_create(
-      email:    auth.info.email,
-      name:     auth.info.name,
-      login:    auth.extra.raw_info.login
+    where(provider: auth.provider, uid: auth.uid).first_or_initialize(
+      email:  auth.info.email,
+      name:   auth.info.name,
+      login:  auth.extra.raw_info.login,
+      token:  auth.credentials.token
     )
   end
+
+  def organisations
+    request = Net::HTTP::Get.new(GITHUB_ORGS_QUERY_URL)
+    request['Authorization'] = "token #{token}"
+    req_options = { use_ssl: GITHUB_ORGS_QUERY_URL.scheme == 'https' }
+
+    response = Net::HTTP.start(GITHUB_ORGS_QUERY_URL.hostname, GITHUB_ORGS_QUERY_URL.port, req_options) do |http|
+      http.request(request)
+    end
+
+    response.code == '200' ? JSON.parse(response.body).map { |a| a['login'].downcase } : []
```

Comment:
What if the response is more than one page?
Github API paginates (with `Link: <https://api.github.com/resource?page=2>; rel="next"` in response headers), how do you handle that?